Automattic/wp-calypso
# Documentation: include E2E documentation in devdocs.

#### Changes proposed in this Pull Request

This PR adds a new section in the devdocs named `Testing` and incorporates links to the E2E documentation in the devdocs.

Key changes:
- improved formatting of the sidebar.
- incorporate links to E2E documentation from the devdocs.

#### Testing instructions

Navigate to https://container-determined-bhabha.calypso.live/devdocs and check out the Testing category.


Closes https://github.com/Automattic/wp-calypso/issues/61182.
